app/code/Magento/CatalogSearch/Block/Result.php

```diff
@@ -47,23 +48,31 @@ class Result extends Template
      */
     private $queryFactory;
 
+    /**
+     * @var Config
+     */
+    protected $catalogConfig;
+
     /**
      * @param Context $context
      * @param LayerResolver $layerResolver
      * @param Data $catalogSearchData
      * @param QueryFactory $queryFactory
+     * @param Config $catalogConfig
      * @param array $data
      */
     public function __construct(
         Context $context,
         LayerResolver $layerResolver,
         Data $catalogSearchData,
         QueryFactory $queryFactory,
+        Config $catalogConfig,
```

Comment:
Can you please inject this as an optional parameter at the end of the `__construct` method. See  https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/index.html#adding-a-constructor-parameter